grouper/log_redact.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+import logging
+import re
+
+@dataclass
+class RedactionPattern:
+    # regular expression to match for redaction
+    regex: str
+    # sensitive string will be replaced by this "redacted_message" in the log output
+    redacted_message: str
+
+class RedactingFormatter(logging.Formatter):
+    """
+    RedactingFormatter masks sensitive regular expressions from log entries.
+    
+    Configure the sensitive log regular expressions in settings.log_redact_patterns.
+    """
+
+    _REDACT_PATTERNS = [
+        RedactionPattern(  # mysql url
+            r"mysql://\S*",
+            "DB_URL_REDACTED",
+        ),
+        RedactionPattern(  # sqlite url
+            r"sqlite://\S*",
+            "DB_URL_REDACTED",
+        ),
+        RedactionPattern(  # anything with password=
+            r"password=\S+",
+            "PASSWORD_REDACTED",
+        ),
+    ]
+
+    def _filter(self, log_string: str) -> str:
+        new_string = log_string
+        for pattern in RedactingFormatter._REDACT_PATTERNS:
+            new_string = re.sub(pattern.regex, f"<{pattern.redacted_message}>", new_string)
```

Comment:
I believe both SQL server and MySQL connection strings are not case sensitive, so you might want to use `flags=re.IGNORECASE` here

Reply:
Good call. Done.